tap_intercom/intercom.py

```diff
@@ -183,12 +177,95 @@ def paginate(
     def call_api(self, url, params={}):
         response = self.SESSION.get(
             url,
-            headers={
-                "Authorization": f"Bearer {self.access_token}",
-                "Accept": "application/json",
-            },
+            headers=self.__make_header(),
+            params=params,
+        )
+        LOGGER.debug(response.url)
+        response.raise_for_status()
+        return response.json()
+
+    @backoff.on_exception(
+        backoff.expo,
+        (
+            requests.exceptions.RequestException,
+            requests.exceptions.HTTPError,
+            ratelimit.exception.RateLimitException,
+            JSONDecodeError,
+        ),
+        max_tries=20,
+        factor=5,
+        max_time=60 * 10,
+        giveup=_is_internal_server_error,
+    )
+    @limits(calls=1000, period=ONE_MINUTE)
+    def call_search_api(self, url, params={}, json=None):
+        response = self.SESSION.post(
+            url,
+            headers=self.__make_header(),
             params=params,
+            json=json,
         )
         LOGGER.debug(response.url)
         response.raise_for_status()
         return response.json()
+
+    def search(self, tap_stream_id, start_date, end_date):
+        url = f"{self.BASE_URL}/{tap_stream_id}/search"
+
+        data_field = tap_stream_id
+        if tap_stream_id == "contacts":
+            data_field = "data"
+
+        # For more options:
+        # https://developers.intercom.com/docs/references/rest-api/api.intercom.io/Contacts/SearchContacts/
+        per_page = 150
```

Comment:
why 150?

Reply:
out of curiosity 👀

Reply:
Tbh I trusted the earlier judgement on what a plausible page size would be. Before it was 150, so I figured I would just keep it as is for now